docs/csharp/programming-guide/statements-expressions-operators/lambda-expressions.md

```diff
@@ -12,276 +12,217 @@ ms.assetid: 57e3ba27-9a82-4067-aca7-5ca446b7bf93
+
+[!code-csharp-interactive[lambda is delegate](~/samples/snippets/csharp/programming-guide/lambda-expressions/Introduction.cs#Delegate)]
+
+You also can assign a lambda expression to an expression tree type:
+
+[!code-csharp-interactive[lambda is expression tree](~/samples/snippets/csharp/programming-guide/lambda-expressions/Introduction.cs#ExpressionTree)]
+
+Or you can pass it directly as a method argument:
+
+[!code-csharp-interactive[lambda is argument](~/samples/snippets/csharp/programming-guide/lambda-expressions/Introduction.cs#Argument)]
+
+When you use method-based syntax to call the <xref:System.Linq.Enumerable.Select%2A?displayProperty=nameWithType> method in the <xref:System.Linq.Enumerable?displayProperty=nameWithType> class (as you do in LINQ to Objects and LINQ to XML) the parameter is a delegate type <xref:System.Func%602?displayProperty=nameWithType>. A lambda expression is the most convenient way to create that delegate. When you call the <xref:System.Linq.Queryable.Select%2A?displayProperty=nameWithType> method in the <xref:System.Linq.Queryable?displayProperty=nameWithType> class (as you do in LINQ to SQL) the parameter type is an expression tree type [`Expression<Func<TSource,TResult>>`](<xref:System.Linq.Expressions.Expression%601>). Again, a lambda expression is just a very concise way to construct that expression tree. The lambdas allow the `Select` calls to look similar although in fact the type of object created from the lambda is different.
+
+All restrictions that apply to [anonymous methods](anonymous-methods.md) also apply to lambda expressions.
   
 ## Expression lambdas
 
- A lambda expression with an expression on the right side of the => operator is called an *expression lambda*. Expression lambdas are used extensively in the construction of [Expression Trees](../concepts/expression-trees/index.md). An expression lambda returns the result of the expression and takes the following basic form:
-  
+A lambda expression with an expression on the right side of the `=>` operator is called an *expression lambda*. Expression lambdas are used extensively in the construction of [expression trees](../concepts/expression-trees/index.md). An expression lambda returns the result of the expression and takes the following basic form:
+
 ```csharp
 (input-parameters) => expression
 ```
 
- The parentheses are optional only if the lambda has one input parameter; otherwise they are required. Two or more input parameters are separated by commas enclosed in parentheses:  
+The parentheses are optional only if the lambda has one input parameter; otherwise they are required.
+
+Specify zero input parameters with empty parentheses:  
+
+[!code-csharp[zero parameters](~/samples/snippets/csharp/programming-guide/lambda-expressions/ExpressionAndStatementLambdas.cs#ZeroParameters)]
```

Comment:
This code include is empty. It could be a problem with the opening and closing tags in the sample.

Reply:
@BillWagner thanks for review and catching those ones. I've overlooked it; fix is dotnet/samples#719